### PR TITLE
[AppSec] Enabled WAF stateful mode (WAF Additive API)

### PR DESCRIPTION
**Pull request by ValentinZakharov**
Changed the way we are calling WAF to enforce stateful mode (using WAF Additive API).
Since we are calling WAF several times at different stages of request processing, we need to keep state of WAF (collected data) between calls. 
For example, we will call WAF three times:
1. At the beginning, when parsed URI and http headers;
2. When request body read;
3. When response status available.

For each stage, we need to keep information about perviously collected data related to current request. This requirement based on fact that some of attacks could be detected by compounded data received at different stages.
Enforce using Additive API, allows to share collected data between WAF calls and will be bounded by request context lifetime.

Basically, we are creating WAF Additive context, during the first WAF call and dispose it when `REQUEST_END` event happens. Given WAF Additive context is kept with the `AppSecRequestContext` and used to process only the data related to current request.

### CODE REVIEW

**Review by bantonsson — COMMENTED** [review]
This looks good, but I would really like there to be some other cleanup mechanism as well, i.e. max outstanding requests and/or time since request start, to clean up the `Additive`, so we don't have unbounded growth.

**Review by bantonsson — APPROVED** [review]
So I thought a bit more about this, and as long as we don't hold on to the `AppSecRequestContext` in any other place, I think that automatic cleanup should work fine. It's a bit unfortunate that the PowerWaf code is using finalizers, but that is not for this PR to clean up.

**Comment by ValentinZakharov** [comment]
But, you are correct, in case if `REQUEST_END` event will not be triggered in some instrumentation, this will lead us to memory leak (because of no explicit call `additive.close()`

**Comment by bantonsson** [comment]
From reading the PowerWaf java bridge code, I think that as long as nobody holds on to the `AppSecRequestContext`, it will be freed by the finalizer in the `Additive`. This is not ideal, but better than always leaking when there is no `REQUEST_END`.
